spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/query/AbstractJdbcQuery.java

```diff
@@ -89,10 +91,14 @@ public JdbcQueryMethod getQueryMethod() {
 			return createModifyingQueryExecutor();
 		}
 
-		if (queryMethod.isCollectionQuery() || queryMethod.isStreamQuery()) {
+		if (queryMethod.isCollectionQuery()) {
 			return extractor != null ? getQueryExecution(extractor) : collectionQuery(rowMapper);
 		}
 
+		if (queryMethod.isStreamQuery()) {
```

Comment:
I don't think we should ignore a configured `ResultSetExtractor`. This would make the actual result in terms of instances returned depend on the container (`Stream` vs. `Collection`).

Instead we should fall back to the existing behaviour for `ResultSetExtractor`.

Reply:
Sorry, I didn't have the time to get back to you on this, but I'll finish this up in the upcoming weeks. Looking forward to getting this merged. 🙂